types/mempool/sender_nonce.go

```diff
@@ -0,0 +1,199 @@
+package mempool
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	huandu "github.com/huandu/skiplist"
+)
+
+var (
+	_ Mempool  = (*senderNonceMempool)(nil)
+	_ Iterator = (*nonceMempoolIterator)(nil)
+)
+
+type senderTxs struct {
+	txQueue *huandu.SkipList
+	head    *huandu.Element
+}
+
+func newSenderTxs() senderTxs {
+	return senderTxs{
+		head:    nil,
```

Comment:
Personally I find the name `head` confusing since this position moves, and head implies static front of list.

suggest: `cursor`, `currentPosition`

Reply:
Ok this makes sense to me will do the change I like cursor